### PR TITLE
Merge toolsHCK into rtoolsHCK to prevent version mismatch

### PR DESCRIPTION
**Pull request by Jedoku**

### CODE REVIEW

**Review by akihikodaki — CHANGES_REQUESTED** [review]
I don't think it is beneficial to have two levels (`shared` and `tools` for toolsHCK. I also don't know what `shared` means.

**Comment by Jedoku** [comment]
> I don't think it is beneficial to have two levels (`shared` and `tools` for toolsHCK. I also don't know what `shared` means.

how about 
`extras/
  - toolsHCK.ps1
  - LICENSE.toolsHCK
  - README.toolsHCK.md`

**Comment by akihikodaki** [comment]
> > I don't think it is beneficial to have two levels (`shared` and `tools` for toolsHCK. I also don't know what `shared` means.
> 
> how about `extras/

`extras` is misleading. They are not extras, but essential part of rtoolsHCK.

> 
>     * toolsHCK.ps1
> 
>     * LICENSE.toolsHCK

`LICENSE.toolsHCK` is redundant and unnecessary.

> 
>     * README.toolsHCK.md`

**Comment by akihikodaki** [comment]
`README.md` also needs to be updated as it refers to toolsHCK in a separate repository.

File: lib/rtoolsHCK.rb [hunk 1]
```diff
@@ -134,7 +134,7 @@ def logger(level, progname = nil, &)
   #                    the local machine
   #                    (default: disabled)
   #   :l_script_file - The toolsHCK.ps1 file path on local machine
-  #                    (default: disabled)
+  #                    (default: shared/tools/toolsHCK.ps1)
```
Comment (akihikodaki) [inline lib/rtoolsHCK.rb]: This needs an update.

I also suggest removing `l_script_file` option and hardcoding the path. We do not want to mix different toolsHCK and rtoolsHCK.

**Review by akihikodaki — CHANGES_REQUESTED** [review]

File: lib/rtoolsHCK.rb [hunk 2]
```diff
@@ -144,6 +142,8 @@ def initialize(init_opts)
     @log_to_stdout = init_opts[:log_to_stdout]
     @stdout_logger = Logger.new($stdout) if @log_to_stdout
     @logger = init_opts[:logger]
+    @l_script_file = File.expand_path('../shared/tools/toolsHCK.ps1', __dir__)
```
Comment (akihikodaki) [inline lib/rtoolsHCK.rb]: It no longer has to be kept as an instance variable of `RToolsHCK`. We can hardcode it at `Server#deploy_script_file`, which actually uses it.

**Review by akihikodaki — CHANGES_REQUESTED** [review]

File: lib/server.rb [hunk 3]
```diff
@@ -38,28 +38,28 @@ def load_instance_variables(init_opts)
...
 
   def deploy_script_file
+    tools_hck = File.expand_path('../tools/toolsHCK.ps1', __dir__)
     logger('debug', 'server/initialize') { 'deploying script file on remote' }
-    unless File.file?(@l_script_file)
+    unless File.file?(tools_hck)
       raise ServerError.new('server/initialize'),
-            "can't find the l_script_file specified."
+            "can't find the file specified."
     end
```
Comment (akihikodaki) [inline lib/server.rb]: Please remove this. We don't have to check the existence of files in gem (so you don't have to check if a file exists when `require`, for example).

File: lib/server.rb [hunk 4]
```diff
@@ -38,28 +38,28 @@ def load_instance_variables(init_opts)
...
     @winrm_fs.delete(@r_script_file)
-    @winrm_fs.upload(File.expand_path(@l_script_file), @r_script_file)
+    @winrm_fs.upload(File.expand_path(tools_hck), @r_script_file)
```
Comment (akihikodaki) [inline lib/server.rb]: This call of `File.expand_path` is no longer necessary.

**Review by akihikodaki — CHANGES_REQUESTED** [review]

**Review by akihikodaki — CHANGES_REQUESTED** [review]
One last request: remove "debuging" from the commit message. It's "debugging", and we don't need it anyway because we want to prevent "version mismatch" itself anyway.

File: README.md [hunk 5]
```diff
@@ -41,7 +41,7 @@ A hash that contains:
 | :log_to_stdout | Log to STDOUT switch<br>**(default: false)**
 | :logger | The ruby logger object for logging<br>**(default: disabled)**
 | :outp_dir | The path of the directory to fetch the output files to on the local machine<br>**(default: disbaled)**
-| :script_file | The toolsHCK.ps1 file path on local machine<br>**(default: disabled)**
+| :script_file | The toolsHCK.ps1 file path on local machine<br>**(default: /tools/toolsHCK.ps1)**
```
Comment (akihikodaki) [inline README.md]: I missed this. This should be removed.

File: lib/rtoolsHCK.rb [hunk 6]
```diff
@@ -144,6 +142,7 @@ def initialize(init_opts)
     @log_to_stdout = init_opts[:log_to_stdout]
     @stdout_logger = Logger.new($stdout) if @log_to_stdout
     @logger = init_opts[:logger]
+
```
Comment (akihikodaki) [inline lib/rtoolsHCK.rb]: This is unrelated.

**Review by akihikodaki — CHANGES_REQUESTED** [review]
I found two minor problems. They should be truely the last things remaining.

**Review by akihikodaki — APPROVED** [review]

**Review by kostyanf14 — APPROVED** [review]

**Pull request merged**